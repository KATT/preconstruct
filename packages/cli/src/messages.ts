import { PKG_JSON_CONFIG_FIELD } from "./constants";

type Field = "main" | "module" | "browser" | "umd:main";

export let errors = {
  noExportStarFromExternalWhenUsingNodeESM:
    'using "export * from" is not allowed with external modules when using the experimental nodeESM flag',
  noSource: (source: string) =>
    `no source file was provided, please create a file at ${source} or specify a custom source file with the ${PKG_JSON_CONFIG_FIELD} source option`,
  deniedWriteMainField: "changing the main field is required to build",
  invalidField: (field: Field) => `${field} field is invalid`,
  umdNameNotSpecified: `the umd:main field is specified but a umdName option is not specified. please add it to the ${PKG_JSON_CONFIG_FIELD} field in your package.json`,
  noEntrypointPkgJson: "There is a missing package.json for an entrypoint",
  noEntrypoints:
    "packages must have at least one entrypoint, this package has no entrypoints",
};

import { createPromptConfirmLoader } from "./prompt";

export let confirms = {
  writeMainField: createPromptConfirmLoader(
    "preconstruct is going to change the main field in your package.json, are you okay with that?"
  ),
  writeModuleField: createPromptConfirmLoader(
    "would you like to generate module builds? this will write to the module field in your package.json"
  ),
  fixModuleField: createPromptConfirmLoader(
    "would you like to fix the module field?"
  ),
  fixUmdBuild: createPromptConfirmLoader(
    "would you like to fix the umd field?"
  ),
  fixBrowserField: createPromptConfirmLoader(
    "would you like to fix the browser build?"
  ),
  createEntrypointPkgJson: createPromptConfirmLoader(
    "A package.json file does not exist for this entrypoint, would you like to create one automatically?"
  ),
  createEntrypoint: createPromptConfirmLoader(
    "This glob does not match anything, would you like to create an entrypoint for it?"
  ),
};

export let inputs = {
  getUmdName: "what should the name used for UMD bundles be?",
  getSource: "what should the source file for this entrypoint be?",
};

export let infos = {
  validField: (field: Field) => `${field} field is valid`,
  validEntrypoint: "a valid entry point exists.",
  validPackageEntrypoints: "package entrypoints are valid",
};

export let successes = {
  validProject: "project is valid!",
  startedWatching: "started watching!",
};
