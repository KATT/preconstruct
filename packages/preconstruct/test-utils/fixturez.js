// @flow
// https://github.com/jamiebuilds/fixturez/
import path from "path";
import * as fsExtra from "fs-extra";
import onExit from "signal-exit";
import _fixturez from "fixturez";
import uuid from "uuid/v4";

function fixturez(cwd: string) {
  let { find } = _fixturez(cwd);
  let created = [];
  function temp(): string {
    let tempDir = path.join(
      __dirname,
      "node_modules",
      "fixture-copies",
      uuid()
    );
    fsExtra.ensureDirSync(tempDir);
    created.push(tempDir);
    return tempDir;
  }

  function copy(name: string): string {
    let dest = path.join(temp(), name);
    fsExtra.copySync(find(name), dest);
    return dest;
  }

  function cleanup() {
    let err;
    created.forEach(tempDir => {
      try {
        fsExtra.removeSync(tempDir);
      } catch (e) {
        err = e;
      }
    });
    created.length = 0;
    if (err) throw err;
  }

  onExit(cleanup);

  return {
    find,
    copy
  };
}

module.exports = fixturez;
