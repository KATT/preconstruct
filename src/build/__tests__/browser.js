// @flow
import build from "../";
import fixturez from "fixturez";
import { snapshotDirectory } from "../../../test-utils";
import { confirms } from "../../messages";

const f = fixturez(__dirname);

jest.mock("../../prompt");

test("two-entrypoints-one-with-typeof-browser-one-without", async () => {
  let tmpPath = f.copy("two-entrypoints-one-with-typeof-window-one-without");

  confirms.addBrowserField.mockReturnValue(Promise.resolve(true));

  await build(tmpPath);

  expect(confirms.addBrowserField).toBeCalledTimes(1);

  await snapshotDirectory(tmpPath, "all");
});
