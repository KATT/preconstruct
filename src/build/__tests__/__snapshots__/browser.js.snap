// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`two-entrypoints-one-with-typeof-browser-one-without: dist/multiple-entrypoints.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});

let sum = (a, b) => a + b;

exports.sum = sum;
"
`;

exports[`two-entrypoints-one-with-typeof-browser-one-without: dist/multiple-entrypoints.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./multiple-entrypoints.cjs.prod.js\\");
} else {
  module.exports = require(\\"./multiple-entrypoints.cjs.dev.js\\");
}
"
`;

exports[`two-entrypoints-one-with-typeof-browser-one-without: dist/multiple-entrypoints.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
let sum = (a, b) => a + b;
exports.sum = sum;
"
`;

exports[`two-entrypoints-one-with-typeof-browser-one-without: dist/multiple-entrypoints.esm.js 1`] = `
"let sum = (a, b) => a + b;

export { sum };
"
`;

exports[`two-entrypoints-one-with-typeof-browser-one-without: multiply/dist/multiple-entrypoints.cjs.dev.js 1`] = `
"'use strict';

Object.defineProperty(exports, '__esModule', {
  value: true
});
let multiply = typeof window === \\"undefined\\" ? (a, b) => a * b : (a, b) => a * b;
exports.multiply = multiply;
"
`;

exports[`two-entrypoints-one-with-typeof-browser-one-without: multiply/dist/multiple-entrypoints.cjs.js 1`] = `
"'use strict';

if (process.env.NODE_ENV === \\"production\\") {
  module.exports = require(\\"./multiple-entrypoints.cjs.prod.js\\");
} else {
  module.exports = require(\\"./multiple-entrypoints.cjs.dev.js\\");
}
"
`;

exports[`two-entrypoints-one-with-typeof-browser-one-without: multiply/dist/multiple-entrypoints.cjs.prod.js 1`] = `
"\\"use strict\\";
Object.defineProperty(exports, \\"__esModule\\", { value: !0 });
let multiply = (a, b) => a * b;
exports.multiply = multiply;
"
`;

exports[`two-entrypoints-one-with-typeof-browser-one-without: multiply/dist/multiple-entrypoints.esm.js 1`] = `
"let multiply = typeof window === \\"undefined\\" ? (a, b) => a * b : (a, b) => a * b;

export { multiply };
"
`;

exports[`two-entrypoints-one-with-typeof-browser-one-without: multiply/package.json 1`] = `
"{
  \\"main\\": \\"dist/multiple-entrypoints.cjs.js\\",
  \\"module\\": \\"dist/multiple-entrypoints.esm.js\\",
  \\"preconstruct\\": {
    \\"source\\": \\"../src/multiply.js\\"
  }
}
"
`;

exports[`two-entrypoints-one-with-typeof-browser-one-without: package.json 1`] = `
"{
  \\"name\\": \\"multiple-entrypoints\\",
  \\"version\\": \\"1.0.0\\",
  \\"main\\": \\"dist/multiple-entrypoints.cjs.js\\",
  \\"license\\": \\"MIT\\",
  \\"private\\": true,
  \\"module\\": \\"dist/multiple-entrypoints.esm.js\\",
  \\"preconstruct\\": {
    \\"source\\": \\"src/sum\\",
    \\"entrypoints\\": [
      \\".\\",
      \\"multiply\\"
    ]
  }
}
"
`;

exports[`two-entrypoints-one-with-typeof-browser-one-without: src/multiply.js 1`] = `
"export let multiply =
  typeof window === \\"undefined\\" ? (a, b) => a * b : (a, b) => a * b;
"
`;

exports[`two-entrypoints-one-with-typeof-browser-one-without: src/sum.js 1`] = `"export let sum = (a, b) => a + b;"`;
